cmake/windows-setup.cmake

```diff
@@ -65,7 +65,7 @@ set(FLB_OUT_RETRY              No)
 set(FLB_OUT_SPLUNK            Yes)
 set(FLB_OUT_STACKDRIVER       Yes)
 set(FLB_OUT_STDOUT            Yes)
-set(FLB_OUT_LIB                No)
+set(FLB_OUT_LIB               Yes)
```

Comment:
@leonardo-albertovich is there any issue with this on Windows?

I know this solves your particular problem @SteRoh but not sure we want to change the default for all. Could we start adding the override support into the CMake configuration so that we keep the current defaults but allow people to override them with a `-DFLB_XXX=` in the call to CMake? If not defined/empty then set the default. 

@niedbalski what do you think? I'm keen to improve the Windows experience and @SteRoh also seems willing!

Reply:
In fact, why are we forcing all these settings explicitly? We should only `set` stuff that definitely needs setting and leave the options in the main `CMakeLists.txt`.

Reply:
I agree with you @patrick-stephens, the best solution would be to make windwos cmake settings settable.

For my ci/cd it was much easier to patch only one file, instead of introduce cmake variables and checks.

Reply:
Sure, I'll take the time to go through this issue tomorrow and then I'll get back to you.

Reply:
Yeah seems fine to me but @leonardo-albertovich can review the specific settings.